modules/imgcodecs/src/grfmt_tiff.cpp

```diff
@@ -1074,6 +1074,9 @@ bool  TiffDecoder::readData( Mat& img )
         else
             cvtColor(img, img, COLOR_XYZ2BGR);
     }
+
+    m_animation.frames.push_back(img);
+
```

Comment:
TIFF is not animation by design. The ImageCollection class is not always used for animations. I propose to return empty animation object, if the source data was not designed for it like TIFF.